tools/include/eosio/gen.hpp

```diff
@@ -683,5 +684,124 @@ struct generation_utils {
 
       return in_kv_namespace && is_internal;
    }
+
+   inline bool is_write_host_func( const std::string& t ) {
+      static const std::set<std::string> write_host_funcs =
+      {
```

Comment:
I would recommend only using the C name for looking for these host functions.
I.e. instead of `eosio::internal_use_do_not_use::set_resource_limits`, just look for the stripped name of `set_resource_limits` and if it is a extern-C function or a regular C function.

The reason for this, is that as you can see, you have to duplicate some of these, also the C++ namespaced names are purely decorative and just supplying context for the smart contract developer.  You should still only have one `set_resource_limits` per smart contract.

Reply:
Fixed. (Method `is_eosio_wasm_import_write_func` is also removed, because when using C name to find write host functions, the updated `is_write_host_func` method can also handle the "eosio_wasm_import" functions. Please let me know if I missed anything. Thanks!)
https://github.com/EOSIO/eosio.cdt/blob/ef00c9d035fcea306131884724691398189be4a3/tools/include/eosio/gen.hpp#L688-L726